compiler/sem/transf.nim

```diff
@@ -174,6 +188,31 @@ proc freshVar(c: PTransf; v: PSym): PNode =
     newVar.owner = owner
     result = newSymNode(newVar)
 
+proc transformDefSym(c: PTransf, n: PNode): PNode =
+  ## Transforms a node that appears in a definition position. If the same
+  ## symbol already appeared in a definition position, a copy is created, and
+  ## a mapping registered so that all further usages of the symbol use the
+  ## copy.
+  # XXX: this is only intended to fix the most simple cases of duplicate symbols
```

Comment:
Ah yes, this would be dangerous to use for routines, for example: forward declaration(s) AST would define the symbol and then it shows up again later in the definition.